File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/viewmodels/SettingsViewModel.kt
package family.aladdin.android.viewmodels

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow

class SettingsViewModel : ViewModel() {
    val isVPNEnabled = MutableStateFlow(true)
    val isNotificationsEnabled = MutableStateFlow(true)
    val isBiometricEnabled = MutableStateFlow(true)
    val protectionLevel = MutableStateFlow(75f)
    
    fun toggleVPN() { isVPNEnabled.value = !isVPNEnabled.value }
    fun toggleNotifications() { isNotificationsEnabled.value = !isNotificationsEnabled.value }
    fun toggleBiometric() { isBiometricEnabled.value = !isBiometricEnabled.value }
    fun updateProtectionLevel(value: Float) { protectionLevel.value = value }
}

File: BACKUP_SCREENS_20251028_200516/SettingsViewModel.swift
import SwiftUI
import Combine

/// ⚙️ Settings View Model
/// Логика для экрана настроек
class SettingsViewModel: ObservableObject {
    
    @Published var isVPNEnabled: Bool = true
    @Published var isNotificationsEnabled: Bool = true
    @Published var isBiometricEnabled: Bool = true
    @Published var protectionLevel: Double = 75
    @Published var selectedLanguage: String = "Русский"
    @Published var selectedTheme: String = "Тёмная"
    @Published var cacheSize: String = "47 MB"
    
    func toggleVPN() {
        isVPNEnabled.toggle()
    }
    
    func toggleNotifications() {
        isNotificationsEnabled.toggle()
    }
    
    func toggleBiometric() {
        isBiometricEnabled.toggle()
    }
    
    func updateProtectionLevel(_ value: Double) {
        protectionLevel = value
    }
    
    func changeLanguage() {
        print("Show language picker")
    }
    
    func changeTheme() {
        print("Show theme picker")
    }
    
    func clearCache() {
        print("Clear cache")
        cacheSize = "0 MB"
    }
    
    func logout() {
        print("Logout user")
    }
}



